services/npm/npm-unpacked-size.service.js

```diff
@@ -0,0 +1,83 @@
+import Joi from 'joi'
+import prettyBytes from 'pretty-bytes'
+import { pathParam, queryParam } from '../index.js'
+import { optionalNonNegativeInteger } from '../validators.js'
+import NpmBase, { packageNameDescription } from './npm-base.js'
+
+const schema = Joi.object({
+  dist: Joi.object({
+    unpackedSize: optionalNonNegativeInteger,
+  }).required(),
+}).required()
+
+export default class NpmUnpackedSize extends NpmBase {
+  static category = 'size'
+
+  static route = this.buildRoute('npm/unpacked-size', { withTag: true })
+
+  static openApi = {
+    '/npm/unpacked-size/{packageName}': {
+      get: {
+        summary: 'NPM Unpacked Size',
+        parameters: [
+          pathParam({
+            name: 'packageName',
+            example: 'npm',
+            description: packageNameDescription,
+          }),
+          queryParam({
+            name: 'registry_uri',
+            example: 'https://registry.npmjs.com',
+          }),
+        ],
+      },
+    },
+    '/npm/unpacked-size/{packageName}/{version}': {
+      get: {
+        summary: 'NPM Unpacked Size (with version)',
+        parameters: [
+          pathParam({
+            name: 'packageName',
+            example: 'npm',
+            description: packageNameDescription,
+          }),
+          pathParam({
+            name: 'version',
+            example: '4.18.2',
+          }),
+          queryParam({
+            name: 'registry_uri',
+            example: 'https://registry.npmjs.com',
+          }),
+        ],
+      },
+    },
+  }
+
+  static defaultBadgeData = { label: 'unpacked size' }
+
+  async fetch({ registryUrl, packageName, version }) {
+    return this._requestJson({
+      schema,
+      url: `${registryUrl}/${packageName}/${version}`,
+    })
+  }
+
+  async handle(namedParams, queryParams) {
+    const { scope, packageName, tag, registryUrl } =
+      this.constructor.unpackParams(namedParams, queryParams)
```

Comment:
I think you had this right first time. We now seem to be mixing the concepts of version and dist tag.
For some badges, we use a tag param, but that's not the same as a version. So for example

latest version of npm package: https://img.shields.io/npm/v/npm
latest version of npm package with dist-tag "next-8": https://img.shields.io/npm/v/npm/next-8

but its not the same as a version. What you're actually implementing here is a version param. i.e: the unpacked size of a specific version of this package. The thing we're accepting here isn't a dist tag.

I'd say just list the actual params in the signature of `handle()` rather than use the wrong helper.

Reply:
Welp, I confused that. Sorry. Alright, let me get to reimplement version back.

Reply:
Completed. Reverted back to using version instead of tags.